bindings/v1/handle_test.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bindings
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/dapr-sandbox/components-go-sdk/internal"
+	contribBindings "github.com/dapr/components-contrib/bindings"
+	proto "github.com/dapr/dapr/pkg/proto/components/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+type recvFakeResp struct {
+	msg *proto.ReadRequest
+	err error
+}
+type fakeTsStream struct {
+	sendCalled   atomic.Int64
+	onSendCalled func(*proto.ReadResponse)
+	sendErr      error
+	recvCalled   atomic.Int64
+	recvChan     chan *recvFakeResp
+}
+
+func (f *fakeTsStream) Send(msg *proto.ReadResponse) error {
+	f.sendCalled.Add(1)
+	if f.onSendCalled != nil {
+		f.onSendCalled(msg)
+	}
+	return f.sendErr
+}
+func (f *fakeTsStream) Recv() (*proto.ReadRequest, error) {
+	f.recvCalled.Add(1)
+	var (
+		resp *proto.ReadRequest
+		err  error
+	)
+	if f.recvChan != nil {
+		fakeResp := <-f.recvChan
+		resp = fakeResp.msg
+		err = fakeResp.err
+	}
+	return resp, err
```

Comment:
I am I not following here: so what hapens if it *is* `nil`, we send those with their zero-value and that's OK?

Reply:
will probably result in a gRPC error because one of those should be not nil, either error or response, but for testing purposes that's ok, I can remove the nil check though